### PR TITLE
Add Stream::zip

### PR DESCRIPTION
**Pull request by tirr-c**

### CODE REVIEW

File: src/stream/stream/zip.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+impl<A: fmt::Debug + Stream, B: fmt::Debug> fmt::Debug for Zip<A, B> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.debug_struct("Zip")
+            .field("first", &self.first)
+            .field("second", &self.second)
+            .finish()
+    }
+}
+
+impl<A: Unpin + Stream, B: Unpin> Unpin for Zip<A, B> {}
```
Comment (stjepang) [inline src/stream/stream/zip.rs]:
```suggestion
impl<A: Unpin, B: Unpin> Unpin for Zip<A, B> {}
```

File: src/stream/stream/mod.rs [hunk 2]
```diff
@@ -545,6 +547,49 @@ pub trait Stream {
...
+    ///
+    /// ```
+    /// # fn main() { async_std::task::block_on(async {
+    /// #
+    /// use std::collections::VecDeque;
+    /// use async_std::stream::Stream;
+    ///
+    /// let l: VecDeque<isize> = vec![1, 2, 3].into_iter().collect();
+    /// let r: VecDeque<isize> = vec![4, 5, 6, 7].into_iter().collect();
+    /// let mut s = l.zip(r);
```
Comment (stjepang) [inline src/stream/stream/mod.rs]: I'm a bit confused - how does this work? Are `l` and `r` streams or `VecDeque`s here?

Comment (tirr-c) [reply src/stream/stream/mod.rs]: `VecDeque`s implement `futures::stream::Stream` so they work as streams.

File: src/stream/stream/zip.rs (hunk 1 above)
Comment (tirr-c) [reply src/stream/stream/zip.rs]: It doesn't work because of `A: Stream` bound on `struct Zip`.

**Comment by yoshuawuyts** [comment]
Ref #129

**Pull request merged**